README.md

```diff
@@ -1,289 +1,590 @@
-
-Extra binding properties **SHOULD** also be defined by the bindable service, using one of the patterns defined in [Pointer to binding data](#pointer-to-binding-data).
-
-The data that is injected or mounted into the container may have a different name because of a few reasons:
-* the backing service may have chosen a different name (discouraged, but allowed).
-* a custom name may have been chosen using the `dataMappings` portion of the `ServiceBinding` CR.
-* a prefix may have been added to certain items in `ServiceBinding`, either globally or per service.  
-
-### Request service binding
-
-Binding is requested by the consuming application, or an entity on its behalf such as the [Runtime Component Operator](https://github.com/application-stacks/runtime-component-operator), via a custom resource that is applied in the same cluster where an implementation of this specification resides.
-
-Since the reference implementation for most of this specification is the [Service Binding Operator](https://github.com/redhat-developer/service-binding-operator) we will be using the `ServiceBinding` CRD, which resides in [this folder](https://github.com/redhat-developer/service-binding-operator/tree/master/deploy/crds), as the entity that holds the binding request.  
-
-**Note** - a few updates / enhancements are being proposed to the current `ServiceBinding` CR, tracked [here](https://github.com/application-stacks/service-binding-specification/issues/16#issuecomment-605309629).
-
-Sample CR
+## Resource Type Schema
+
+```yaml
+status:
+  binding:
+    name:  # string
+```
+
+## Example Resource
+
+```yaml
+...
+status:
+  ...
+  binding:
+    name: production-db-secret
 ```
+
+## Well-known Secret Entries
+
+Other than the required `kind` entry and the recommended `provider` entry, there are no other reserved `Secret` entries.  In the interests of consistency, if a `Secret` includes any of the following entry names, the entry value **MUST** meet the specified requirements:
+
+| Name | Requirements
+| ---- | ------------
+| `host` | A DNS-resolvable host name or IP address
+| `port` | A valid port number
+| `uri` | A valid URI as defined by [RFC3986](https://tools.ietf.org/html/rfc3986)
+| `username` | A string-based username credential
+| `password` | A string-based password credential
+| `certificates` | A collection of PEM-encoded X.509 certificates, representing a certificate chain used in mTLS client authentication
+| `privateKey` | A PEM-encoded private key used in mTLS client authentication
+
+`Secret` entries that do not meet these requirements **MUST** use different entry names.
+
+# Application Projection
+
+A Binding `Secret` **MUST** be volume mounted into a container at in `$SERVICE_BINDINGS_ROOT/<binding-name>` with directory names matching the name of the binding.  Binding names **MUST** match `[a-z0-9\-\.]{1,253}`.  The `$SERVICE_BINDINGS_ROOT` environment variable **MUST** be declared and can point to any valid file system location.
```

Comment:
```suggestion
A Binding `Secret` **MUST** be volume mounted into a container at `$SERVICE_BINDINGS_ROOT/<binding-name>` with directory names matching the name of the binding entry names.  Binding entry names **MUST** match `[a-z0-9\-\.]{1,253}`.  The `$SERVICE_BINDINGS_ROOT` environment variable **MUST** be declared and can point to any valid file system location.
```

Reply:
Is this requirement talking about the name of the binding, the name of keys within the binding, or both?

Reply:
The former.  We _should_ police the names of the entries within each binding, but there's no documentation of that at this time.

Reply:
ok, then scratch the 2nd part of my suggestion, but the first part I believe is still accurate: the file names are the entry names, whereas the binding name is the directory. Correct?

Reply:
I think that this change should just be discarded and the change is to actually introduce a _new_ section that describes the entry restrictions.  And really those are the `Secret` name restrictions.

Reply:
Addressed in 2731c8a